### PR TITLE
[WIP] feat:Add support for project description

### PR DESCRIPTION
**Pull request by shadmansaleh**
Added project description meta data.

This is suppose to solves [this issue](https://github.com/shadmansaleh/lualine.nvim/pull/45#discussion_r707817209)

### Note: This is untested

@kdheepak how do you test this stuff :P ?

### CODE REVIEW

**Comment by kdheepak** [comment]
Currently untested! I just add sections and text to the `doc/panvimdoc.md` corresponding to the code I add. I’ll have to add a test to check that the output is what I expect.

**Comment by shadmansaleh** [comment]
that's not what I meant. How do you check if the workflow is broken or not? Or if it's emitting correct output on local machine ? Run the entrypoint,sh ?

File: scripts/panvimdoc.lua [hunk 1]
```diff
@@ -101,40 +101,44 @@ function Doc(body, metadata, variables)
...
+    local r = "Last change: " .. date
+    local n = math.max(0, 78 - #vim_doc_title_tag - #m - #r)
+    local s = string.rep(" ", math.floor(n / 2))
+    project_description = s .. m .. s .. r
+  end
+  if mod(math.max(0, 78 - #vim_doc_title_tag - #project_description), 2) == 1 then
+    add(vim_doc_title_tag.." "..project_description)
   else
-    add(l .. s .. m .. s .. r)
+    add(vim_doc_title_tag..project_description)
```
Comment (kdheepak) [inline scripts/panvimdoc.lua:141]:
```suggestion
padding = string.rep(" ", 78 - #vim_doc_title_tag - #project_description)
add(vim_doc_title_tag..padding..project_description)
```

**Comment by kdheepak** [comment]
Ah. I have the repository to run the auto commit when it is merged to `main`. 

https://github.com/kdheepak/panvimdoc/blob/8a478da661907aef1561ea2b0f720f3b978fe6d3/.github/workflows/panvimdoc.yml#L9-L19

When we merge this, we'll know the output. We can change that to work on any branch if you'd like.

**Comment by kdheepak** [comment]
When a commit is made to `main` it'll auto commit like this one: https://github.com/kdheepak/panvimdoc/commit/8a478da661907aef1561ea2b0f720f3b978fe6d3

**Comment by kdheepak** [comment]
If you have `pandoc` installed, you can run `entrypoint.sh` with the right arguments or just run `pandoc` manually. Something like:

```sh
$ pandoc -f markdown+link_attributes --metadata=project:panvimdoc --lua-filter scripts/skip-blocks.lua --lua-filter scripts/include-files.lua -t scripts/panvimdoc.lua README.md -o doc/panvimdoc.txt
```

**Comment by kdheepak** [comment]
I'm back at my computer now. I can take it from here if you don't mind.

File: scripts/panvimdoc.lua (hunk 1 above)
Comment (shadmansaleh) [reply scripts/panvimdoc.lua]: this works too right ?
da85ea34f2adf56b2300f3cf2c9ea3e90918decf

**Comment by shadmansaleh** [comment]
> I'm back at my computer now. I can take it from here if you don't mind.

not at all.

**Comment by shadmansaleh** [comment]
> When we merge this, we'll know the output.

Alright.

> We can change that to work on any branch if you'd like.

won't be required I think.

> If you have pandoc installed, you can run entrypoint.sh with the right arguments

ok let me install it

File: scripts/panvimdoc.lua (hunk 1 above)
Comment (kdheepak) [reply scripts/panvimdoc.lua]: Yes!

**Comment by kdheepak** [comment]
I think this looks good! I tested it locally and it works well :)

**Comment by kdheepak** [comment]
I tried to get the autocommit to work on PRs, but doesn't seem to work on `pull_request` events correctly. I'll debug that later.

**Comment by kdheepak** [comment]
https://github.com/kdheepak/panvimdoc/commit/a39293f6bf194c81e7159b00c28f0075d20a1b15

**Pull request merged**